### PR TITLE
PIM-8169: Start benching product

### PR DESCRIPTION
**Pull request by anaelChardan**
**Description (for Contributor and Core Developer)**

This PR introduces a new script to be able to bench the product API on your own.
It is the first step, a baby step and we'll keep improving it.

It benches our reference catalog and takes 26 min.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: bin/benchmark_products_api.sh [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+if [ $# -eq 0 ]; then
+    REFERENCE_CATALOG_FILE="$CONFIG_PATH/product_api_catalog.yml"
+else
+    if [ ! -f "$CONFIG_PATH/$1" ]; then
+        echo >&2 "The file does not exist"; exit 1;
+    fi;
+    REFERENCE_CATALOG_FILE="$CONFIG_PATH/$1"
+fi;
+
+PUBLIC_PIM_HTTP_PORT="8081"
```
Comment (mmetayer) [inline bin/benchmark_products_api.sh]: Is this line necessary? (I kinda remember it *is*, but I can't figure out why..)

File: tests/benchmarks/test_product_api_catalog.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+catalog:
```
Comment (mmetayer) [inline tests/benchmarks/test_product_api_catalog.yml]: this one is not used

File: bin/benchmark_products_api.sh (hunk 1 above)
Comment (anaelChardan) [reply bin/benchmark_products_api.sh]: Don't know remember neither :/

File: tests/benchmarks/product_api_catalog.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            count: 50
+        attributes:
+            count: 550
+            percentage_of_localizable: 12
+            percentage_of_scopable: 3
+            percentage_of_localizable_and_scopable: 6
+        families:
+            count: 100
+            attributes_count: 120
+        products:
```
[the remark this answers is not in the document]
Comment (anaelChardan) [reply tests/benchmarks/product_api_catalog.yml]: nop

**Pull request merged**